crates/primitives/src/header.rs

```diff
@@ -931,18 +837,97 @@ impl SealedHeader {
 impl proptest::arbitrary::Arbitrary for SealedHeader {
     type Parameters = ();
     fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
-        use proptest::prelude::{any, Strategy};
+        prop_compose! {
+            fn valid_header_strategy()(
+                mut header in any::<Header>(),
+                eip_4844_active in any::<bool>(),
+                blob_gas_used in any::<u64>(),
+                excess_blob_gas in any::<u64>(),
+                parent_beacon_block_root in any::<B256>()
+            ) -> SealedHeader {
+                // EIP-1559 logic
+                if header.base_fee_per_gas.is_none() {
+                    // If EIP-1559 is not active, clear related fields
+                    header.withdrawals_root = None;
+                    header.blob_gas_used = None;
+                    header.excess_blob_gas = None;
+                    header.parent_beacon_block_root = None;
+                }else {
+                    // 4895 is not active
+                    if header.withdrawals_root.is_none() {
+                        header.blob_gas_used = None;
+                        header.excess_blob_gas = None;
+                        header.parent_beacon_block_root = None;
+                    } else {
+                        // 4895 is active
+                        if eip_4844_active {
+                            // EIP-4844 is active
+                            header.blob_gas_used = Some(blob_gas_used);
+                            header.excess_blob_gas = Some(excess_blob_gas);
+                            header.parent_beacon_block_root = Some(parent_beacon_block_root);
+                        } else {
+                            // EIP-4844 is not active
+                            header.blob_gas_used = None;
+                            header.excess_blob_gas = None;
+                            header.parent_beacon_block_root = None;
+                        };
+                    }
+                }
 
-        any::<(Header, BlockHash)>().prop_map(move |(header, _)| header.seal_slow()).boxed()
-    }
+                // Seal the header
+                header.seal_slow()
+            }
+        }
 
+        valid_header_strategy().boxed()
+    }
     type Strategy = proptest::strategy::BoxedStrategy<SealedHeader>;
 }
 
-#[cfg(any(test, feature = "arbitrary"))]
+use proptest::prelude::*;
```

Comment:
imports should always be at the top of the file, and implementations still need `cfg(any(test, feature = "arbitrary"))`. import may also need to be cfg gated